docs/docs/api_docs/dataframe_lib.md

```diff
@@ -0,0 +1,553 @@
+# Bodo Pandas API (Bodo DataFrame Library)
+
+The Bodo DataFrame Library is designed to accelerate and scale Pandas workflows with just a one-line change — simply replace:
+
+``` py
+import pandas as pd
+```
+
+with
+
+``` py
+import bodo.pandas as pd
+```
+
+and your existing code can immediately take advantage of high-performance, scalable execution.
+
+Key features include:
+
+- __Full Pandas compatibility__ with a transparent fallback mechanism to native Pandas,
+ensuring that your workflows continue uninterrupted even if a feature is not yet supported.
+
+- __Advanced query optimization__ such as
+ filter pushdown, column pruning and join reordering behind the scenes.
+
+- __Scalable MPI-based execution__, leveraging High-Performance Computing (HPC) techniques for efficient parallelism;
+whether you're working on a laptop or running jobs across a large cloud cluster.
+
+- __Vectorized execution__ with streaming and spill-to-disk capabilities,
+making it possible to process datasets larger than memory reliably.
+
+!!! warning
+    Bodo DataFrame Library is under active development and is currently considered experimental.
+    Some features and APIs may not yet be fully supported.
+    We welcome your feedback — please join our community [Slack](https://bodocommunity.slack.com/join/shared_invite/zt-qwdc8fad-6rZ8a1RmkkJ6eOX1X__knA#/shared-invite/email) or open an issue on [our GitHub](https://github.com/bodo-ai/Bodo)
+    if you encounter any problems!
+
+## Lazy Evaluation and Fallback to Pandas
+
+Bodo DataFrame Library operates with lazy evaluation to allow query optimization, meaning operations are recorded into a query plan rather than executed immediately.
+Execution is automatically triggered only when results are actually needed, such as when displaying the first 5 rows in a DataFrame with `print(df.head(5))`.
```

Comment:
Minor: change `head` example to just printing the dataframe? `head` creates a separate plan and is not the same dataframe being materialized.